src/bt-hf-io.c

```diff
@@ -257,11 +314,16 @@ dtmf_cmd_cb(char tone)
 }
 
 static void
+#if ANDROID_VERSION >= 21
+nrec_cmd_cb(bthf_nrec_t nrec, bt_bdaddr_t* bd_addr)
+#else
 nrec_cmd_cb(bthf_nrec_t nrec)
+#endif
 {
   struct pdu_wbuf* wbuf;
 
-  wbuf = create_pdu_wbuf(1, /* NREC */
+  wbuf = create_pdu_wbuf(1 + /* NREC */
+                         6, /* address (BlueZ 5.25) */
```

Comment:
The overall idea is to pick one protocol and go with that for a specific version of B2G. I suggest to simply pick 5.27, go through the code again and check for differences. The only difference between 5.26 and 5.27 seems to be the addition of |ConfigureWBS|. [1]

We can update the protocol for m-c and master branches without problems. But as soon as a version shipped to vendors, or worse end users, we should consider it stable. That means that we should try to get version 2.2 as complete as possible before feature-landing ends in late February.

[1] https://git.kernel.org/cgit/bluetooth/bluez.git/commit/android/hal-ipc-api.txt?id=e0b7cd73531eda84e4ec7362754a9a4f7b760c37